app/models/term.rb

```diff
@@ -1,2 +1,8 @@
 class Term < ApplicationRecord
+  def self.log_query(q)
```

Comment:
add `return if q.blank?` to prevent saving empty queries.
Also clear leading and trailing whitespaces from query with [String#strip](https://apidock.com/ruby/String/strip):

```ruby
def self.log_query(q)
  return if q.blank?
  q = q.strip
  term = Term.find_or_initialize_by(name: q)
  term.entries_count = Entry.search(q).count
  term.popularity += 1  if term.persisted?
  term.save
end
```